### PR TITLE
feat(celery): Set "messaging.system" on span

### PR DESCRIPTION
**Pull request by szokeasaurusrex**
Depends on:
  - #3015
  - #2971
  - #2984

ref #2951

### CODE REVIEW

File: tests/integrations/celery/test_celery.py [hunk 1]
```diff
@@ -704,3 +704,21 @@ def task(): ...
...
     assert span["data"]["messaging.message.retry.count"] == 3
+
+
+@pytest.mark.parametrize("system", ("redis", "amqp"))
+def test_messaging_system(system, init_celery, capture_events):
+    celery = init_celery(enable_tracing=True)
+    events = capture_events()
+
+    # Does not need to be a real URL, since we use always eager
+    celery.conf.broker_url = f"{system}://example.com"  # noqa: E231
```
Comment (szokeasaurusrex) [inline tests/integrations/celery/test_celery.py:715]: Flake8 is giving a false positive E231 on the `:` in the string when running in CI

**Review by edwardgou-sentry — APPROVED** [review]
Worked for me in testing

File: tests/integrations/celery/test_celery.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def init_celery(sentry_init, request):
     def inner(propagate_traces=True, backend="always_eager", **kwargs):
         sentry_init(
             integrations=[CeleryIntegration(propagate_traces=propagate_traces)],
-            **kwargs
+            **kwargs,
```
[the remark this answers is not in the document]
Comment (antonpirker) [reply tests/integrations/celery/test_celery.py:31]: yes, I had that too at some point!

**Review by antonpirker — APPROVED** [review]
lgtm

**Pull request merged**